code/game/objects/items/devices/portable_recharger.dm

```diff
@@ -0,0 +1,337 @@
+			cell.use(charging_cell.charge * seconds_per_tick / 2)
+			using_power = TRUE
+		update_appearance()
+
+	if(istype(charging, /obj/item/ammo_box/magazine/recharge)) //if you add any more snowflake ones, make sure to update the examine messages too.
+		var/obj/item/ammo_box/magazine/recharge/power_pack = charging
+		if(power_pack.stored_ammo.len < power_pack.max_ammo)
+			power_pack.stored_ammo += new power_pack.ammo_type(power_pack)
+			cell.use(charging_cell.charge)
+			using_power = TRUE
+		update_appearance()
+		return
+	if(!using_power && !finished_recharging) //Inserted thing is at max charge/ammo, notify those around us
+		finished_recharging = TRUE
+		playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
+		say("[charging] has finished recharging!")
+
+/obj/item/portable_recharger/emp_act(severity)
+	. = ..()
+	if (. & EMP_PROTECT_CONTENTS)
+		return
+
+	if(ishuman(loc))
+		var/mob/living/carbon/human/holder = loc
+		if(prob(malf_chance) || !EMP_PROTECT_SELF)
+			malfunction(holder)
+			return
+
+	cell?.emp_act(severity)
+
+	if(istype(charging, /obj/item/gun/energy))
+		var/obj/item/gun/energy/energy_gun = charging
+		energy_gun?.cell.emp_act(severity)
+
+	else if(istype(charging, /obj/item/melee/baton/security))
+		var/obj/item/melee/baton/security/batong = charging
+		batong?.cell.charge = 0
+
+/obj/item/portable_recharger/update_overlays()
+	. = ..()
+
+	var/icon_to_use = "[base_icon_state]-[isnull(charging) ? "empty" : (using_power ? "charging" : "full")]"
+	if(hit)
+		. += mutable_appearance(icon, "[base_icon_state]-hit", alpha = src.alpha)
+		. += emissive_appearance(icon, "[base_icon_state]-hit", src, alpha = src.alpha)
+		return
+	. += mutable_appearance(icon, icon_to_use, alpha = src.alpha)
+	. += emissive_appearance(icon, icon_to_use, src, alpha = src.alpha)
+
+/obj/item/portable_recharger/belt
+	name = "belt recharger"
+	icon_state = "beltcharger"
+	base_icon_state = "beltcharger"
+	desc = "A portable belt charging dock for energy based weaponry, PDAs, and other devices. Due its design its slightly more volatile."
+	inhand_icon_state = "beltcharger"
+	lefthand_file = 'icons/mob/inhands/equipment/security_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/security_righthand.dmi'
+	slot_flags = ITEM_SLOT_BELT
+	malf_chance = 10 //More convient to wear, but riskier
+
+/obj/item/portable_recharger/belt/loaded/Initialize(mapload)
+	. = ..()
+	cell = new /obj/item/stock_parts/cell/high(src)
+
+/obj/item/portable_recharger/belt/badass
+	name = "badass belt recharger"
+	icon_state = "badasscharger"
+	base_icon_state = "badasscharger"
+	desc = "A portable belt charging dock for energy based weaponry, PDAs, and other devices. This one will allow you to spin your guns."
+	inhand_icon_state = "badasscharger"
+	alternate_worn_layer = UNDER_SUIT_LAYER
+
+/obj/item/portable_recharger/belt/badass/equipped(mob/user, slot, initial)
+	. = ..()
+	ADD_TRAIT(user, TRAIT_GUNFLIP, CLOTHING_TRAIT)
```

Comment:
This gives the trait if it's held in hands, not sure if you care

Reply:
Because there is a return at the <equipped> for the base portable recharger, it is only when worn